datadog/client/client_tcp.go

```diff
@@ -130,32 +130,23 @@ func (c *ClientTCP) PostLogEntry(logEntry interface{}) (err error) {
 
 	// keep retrying to send and after each retry... disconnect and reconnect to the datadog api
 	for numRetries <= maxRetries {
-		numRetries++
+		if _, err = c.conn.Write([]byte(logEntryJSONStr)); err == nil {
+			break
+		}
 
```

Comment:
Perhaps you should have a backoff here as this has the potential to spin out pretty fast.

Reply:
sounds reasonable